add system variable 'tidb_enable_clustered_index' for v5.0 (#3907)
pingcap/docs: system-variables.md

- Default value: 0
- This variable is used to control whether to enable the cascades planner, which is currently considered experimental.

> ### tidb_enable_clustered_index <!-- New in 5.0 -->

Reviewer: ```suggestion
### tidb_enable_clustered_index <span class="version-mark">Upcoming in v5.0</span>
```

> ### tidb_enable_clustered_index <span class="version-mark">Upcoming in v5.0</span>

Reviewer: ```suggestion
### tidb_enable_clustered_index <!-- New in v5.0 -->
```

- Scope: SESSION | GLOBAL
- Default value: 1

> - This variable is used to control whether to enable the Clustered Index feature.

Reviewer: ```suggestion
- This variable is used to control whether to enable the clustered index feature.
```

>     - This feature is only applicable to newly created tables and has no effect on old tables that have been created.

Reviewer: ```suggestion
    - This feature is only applicable to newly created tables and does not affect the existing old tables.
```

>     - This feature is only applicable to tables whose primary key is the single-column non-integer type or the multi-column type. It does not affect the tables without a primary key and tables with the primary key of the single-column non-integer type.

Reviewer: ```suggestion
    - This feature is only applicable to tables whose primary key is the single-column non-integer type or the multi-column type. It does not affect the tables without a primary key or tables with the primary key of the single-column non-integer type.
```

>     - This feature is only applicable to tables of single column non-integer primary key and multi-column primary key. It does not affect tables without primary keys and tables with single column integer primary keys.

Reviewer: ```suggestion
    - This feature is only applicable to tables whose primary key is the single-column non-integer type or the multi-column type. It does not affect tables without a primary key and tables with the primary key of the single-column non-integer type.
```

>     - You can execute `select tidb_pk_type from information_schema.tables where table_name ='{table_name}'` to check whether a table uses the Clustered Index feature.

Reviewer: ```suggestion
    - You can execute `select tidb_pk_type from information_schema.tables where table_name ='{table_name}'` to check whether the clustered index feature has been enabled on a table.
```

> - After you enable this feature, rows are stored directly on the primary key instead of the rows_id allocated within the system, and use the extra created primary key index to point to row_id.

Reviewer: ```suggestion
- After you enable this feature, rows are stored directly on the primary key instead of on the internally allocated `rows_id` and using the extra primary key index created to point to `row_id`.
```

>     The impact of this feature on performance is mainly reflected in the following aspects:

Reviewer: ```suggestion
    This feature impacts performance in the following aspects:
```

>     - When inserting this variable, each row reduces the write of one index key.

Reviewer: ```suggestion
    - For each `INSERT` operation, there is one less index key written into each row.
```

>     - When using the primary key as the equivalent condition query, you can save one read request.

Reviewer: ```suggestion
    - When you make a query using the primary key as the equivalent condition, one read request can be saved.
```

>     - When using a single-column primary key as a range condition query, you can save multiple read requests.

Reviewer: ```suggestion
    - When you make a query using the primary key as the range condition, multiple read requests can be saved.
```

>     - When using the prefix of a multi-column primary key as an equivalent or range condition query, you can save multiple read requests.

Reviewer: ```suggestion
    - When you make a query using the prefix of the multi-column primary key as the equivalent condition or range condition, multiple read requests can be saved.
```
